apps/keyboard/js/views/swipeable_page_view.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+/* global LayoutPageView, KeyView, SwipeablePanelView, SwipeableSectionView */
+
+(function(exports) {
+/**
+ * SwipeablePageView is used to render a swipeable page,
+ * like the pages in emoji layout.
+ */
+
+function SwipeablePageView() {
+  LayoutPageView.apply(this, arguments);
+}
+
+SwipeablePageView.prototype = Object.create(LayoutPageView.prototype);
+
+SwipeablePageView.prototype.render = function() {
+  var layout = this.layout;
+
+  var container = document.createElement('div');
+  if (this.options.classNames) {
+    container.classList.add.apply(container.classList, this.options.classNames);
+  }
+
+  if (layout.specificCssRule) {
+    container.classList.add(layout.layoutName);
+  }
+
+  var swipeablePanel = new SwipeablePanelView({}, {
+    totalWidth: this.options.totalWidth,
+    itemRenderer: new SwipeableSectionView(this.layout, {}, this.viewManager)
```

Comment:
I also don't know why there is only one `SwipeableSectionView` instance. The division of work between the two is entirely incorrect. Please explain.